src/modals/EditStakeModal.tsx

```diff
@@ -13,59 +15,84 @@ import FormModal, {
     WingedLabelWrap,
 } from '~/modals/FormModal'
 import Label from '~/shared/components/Ui/Label'
-import { toBN } from '~/utils/bn'
+import { BN, toBN } from '~/utils/bn'
 import { fromDecimals, toDecimals } from '~/marketplace/utils/math'
 import { Alert } from '~/components/Alert'
 import { useConfigValueFromChain } from '~/hooks'
+import { ParsedOperator } from '~/parsers/OperatorParser'
+import { useSponsorshipTokenInfo } from '~/hooks/sponsorships'
+import { SponsorshipPaymentTokenName } from '~/components/SponsorshipPaymentTokenName'
+import { isTransactionRejection, sameBN, waitForIndexedBlock } from '~/utils'
+import { ParsedSponsorship } from '~/parsers/SponsorshipParser'
+import { getSponsorshipStakeForOperator } from '~/utils/sponsorships'
+import {
+    forceUnstakeFromSponsorship,
+    reduceStakeOnSponsorship,
+    stakeOnSponsorship,
+} from '~/services/sponsorships'
+import { confirm } from '~/getters/confirm'
+import { Layer } from '~/utils/Layer'
 
-interface Props extends Omit<FormModalProps, 'canSubmit' | 'onSubmit'> {
-    onSubmit: (
-        amountWei: string,
-        differenceWei: string,
-        forceUnstake?: boolean,
-    ) => void | Promise<void>
-    onResolve?: (result: { amountWei: string; differenceWei: string }) => void
-    operatorBalance: string
-    tokenSymbol: string
-    decimals: number
-    currentStake: string
-    leavePenalty: string
-    minLeaveDate: string
-    hasUndelegationQueue: boolean
+interface Props extends Pick<FormModalProps, 'onReject'> {
+    leavePenaltyWei: BN
+    onResolve?: () => void
+    operator: ParsedOperator
+    sponsorship: ParsedSponsorship
 }
 
-export default function EditStakeModal({
-    title = 'Edit stake',
+const DefaultCurrentAmount = toBN(0)
+
+function EditStakeModal({
+    leavePenaltyWei,
     onResolve,
-    onSubmit,
-    operatorBalance: operatorBalanceProp,
-    tokenSymbol = 'DATA',
-    currentStake: currentStakeProp,
-    decimals = 18,
-    leavePenalty: leavePenaltyWei,
-    minLeaveDate,
-    hasUndelegationQueue,
+    onReject,
+    operator: { dataTokenBalanceWei: operatorBalance, id: operatorId, queueEntries },
+    sponsorship: { id: sponsorshipId, stakes, minimumStakingPeriodSeconds },
     ...props
 }: Props) {
     const [busy, setBusy] = useState(false)
 
-    const operatorBalance = toBN(operatorBalanceProp)
+    const { decimals = 18 } = useSponsorshipTokenInfo() || {}
+
+    const stake = useMemo(
+        () => getSponsorshipStakeForOperator(stakes, operatorId),
+        [stakes, operatorId],
+    )
+
+    useEffect(() => {
+        if (!stake) {
+            onReject?.(new Error('Cannot edit. Operator has no stake in sponsorship.'))
```

Comment:
This will dismiss the modal if their stake isn't present. Usually we protect the user from this pathway by disabling the button that'd get them here.